### ISSUE TITLE
Support for teams (in some way)

### ISSUE THREAD

**Issue by almarklein** [almarklein/timetagger#142]
TimeTagger is currently aimed at individuals. This allows it to be lean, because any features specific to teams is left out. This issue is to collect ideas for better teams support, and to set expectations.

I think that some sort of support for using TimeTagger in teams is feasible, but only in a way that maintains the simple/lean UI. Don't expect extensive "manager options", but think more in the line of collaboration as equal members in a team.

Ideas (brainstorm-like):

* Enable sharing a subset of tags via a secret link. You can email that link to someone and they could see the work on those tags in a read-only way. Send it to your co-workers, manager, or client. Some more details in #118.
* Allow more users on the same account. Maybe the timeline should support multiple "tracks".
* ...

**Comment by dzmitry-lahoda** [comment]
timeline with multiple tracks will make things harder for me as I envision a timeline. for now the timeline is simple and good - I am one and one timeline.  additionally, it fits better small screens. what can be done is different filters and visuals for not selected tags (like not `include` tag, but `exclude` tag filter; and not grey out, but invisible).

**Comment by almarklein** [comment]
> timeline with multiple tracks will make things harder for me as I envision a timeline.

Yeah, also it does not scale to beyond 4 or so, so this was probably a bad idea :)

**Comment by boehs** [comment]
I was just thinking about the first:

> Enable sharing a subset of tags via a secret link. You can email that link to someone and they could see the work on those tags in a read-only way. Send it to your co-workers, manager, or client.

Probably also not too hard to implement.

**Comment by almarklein** [comment]
From someone interested in this feature:

> I want to track the time invested in each project 
> (joint hours of all team members - the total sum and per team member) 
> and the total working hours of each team member. [...] if your 
> “tags” can be managed from an admin account with some basic reporting 
> capabilities that will work for us.

**Comment by dzmitry-lahoda** [comment]
Every tool which adds manager/admin makes tool no got for personal use. 

So approach could be to ad allowance to consume some specific streams of my work by set of tags , e.g. foo.bar@baz.com can consume items tagged in `#baz AND #work`.  Then there is separate view which will show timelines like not, but readonly with many columns. So less modifications to existing UI and similar to usual calendars sharing.

So if company wants to work with me, we settle on set of tags and I use these per person hired. If somebody will force tags on me, it will kill my own personal tag system.

**Comment by dzmitry-lahoda** [comment]
So basically I just hate https://github.com/almarklein/timetagger/issues/142#issuecomment-1246426415

**Comment by almarklein** [comment]
First off, in this issue I try to get an idea of what support for teams can look like. Every person and team has (slightly?) different requirements and workflows. What works for you may not work for others, and vice versa.

> So less modifications to existing UI and similar to usual calendars sharing.

I agree with you here, and also about that support for teams should not affect the 1-user UX. But I think what's mentioned in that comment does not contradict that. E.g. the company / team leader could define a tag, and then be able to view all records that include that tag.

**Comment by dzmitry-lahoda** [comment]
`company / team leader could define a tag` so I have tag `foobar` and they define `foobar` -> conflict?

**Comment by dzmitry-lahoda** [comment]
as soon as there is conflict defined by employer, tool is not longer mine way. also one employer told foobar and other foobar tell other thing - same tag. 

may say - each client/contractor/employer will tell me "name_my_company_like" that. 

but what either names be short or without Inc or use tickers from trading exchanges for their names? 

or split my work in single company by teams somehow.

the way I can tell team lead - all work for you goes via this query of tags. that it.

**Comment by dzmitry-lahoda** [comment]
so i have my own brain in tags as in browser/anki/file system symbol links/timetager. if best case tags asked by team fit well to my system, but likely not. so as timetager user I would like to have tool which allows me in simplest case rename tags in stream to team. so really it can be 1. setup tag filter to detect. 2. setup tag renamings and tag merges or tag splits.

**Comment by dzmitry-lahoda** [comment]
@louis-lau thank for intresting feedback 

> Tags for things like projects would need to be managed centrally for the team to ensure consistency

this kills purpose of timetagger as effective tool allowing to mix personal time with one or more customers.

there are other centralized tools on market for you exists for sure :)

>  There's also a "participate" button that copies the task from that user to me, which is great for things like meetings!

what you mean by copies tags? will it mix with my tags? how centralized manager will start tracking copied tasks?


So to tbh something like timetagger will never became centralized hierarchical tool eliminating all flex of tags :)

**Comment by louis-lau** [comment]
> this kills purpose of timetagger as effective tool allowing to mix personal time with one or more customers.

Not really. Only certain tags would be shared.

> what you mean by copies tags? will it mix with my tags? how centralized manager will start tracking copied tasks?

I said task, not tags. Say I have a meeting with #client1, and my colleague Dave is also there, they could just copy that event to their timeline.

> will never became centralized hierarchical tool eliminating all flex of tags

I didn't suggest that :). Just outlined how we use other tools as a team to provide feedback. I would just need a list of predefined tags shared with the team. I think that makes sense? Otherwise how would you actually use it as a team?

**Comment by dzmitry-lahoda** [comment]
thing is as soon as some manager centralizes and forces tags on me,

it immediately conflicts with my tags.

**Comment by dzmitry-lahoda** [comment]
task is just set of predefined tags

**Comment by dzmitry-lahoda** [comment]
i tried to describe it earlier. 

but in short, i peek what tags to share, and it is job of manager to make tags map to team tags.

or tag mapping can be aligned collaboratively.

or even on export, view.

imagine some noisy customer with all task super severe and of high prio.

in my tags i will map all one ladder less. this is healthy.

**Comment by dzmitry-lahoda** [comment]
@louis-lau i used some tools before.

 your description of flows seem like fits non tag first tools.

so imho tag first tool could have go other way, no centralized tag managment.

**Comment by louis-lau** [comment]
Sure but that's just like, your opinion. Just because support for teams could be added and I outlined my usecases for it doesn't mean it would be forced upon you. What are you talking about?

**Comment by almarklein** [comment]
@louis-lau thanks for explaining your use-case. That's very valuable to determine the direction TimeTagger could take.

@dzmitry-lahoda I understand your concern, but I can promise you that if/when I add support for teams, it won't affect the personal experience. 

I don't know yet what it will look like, but I imagine it will feel much more as collaborative. A bit like you have a GH account for your personal projects and you use that same account to contribute to your employer's / client's projects.
